thirdeye/thirdeye-frontend/app/actions/dimensions.js

```diff
@@ -0,0 +1,136 @@
+import { type } from './utils';
+import { COMPARE_MODE_MAPPING } from './constants';
+import fetch from 'fetch';
+import moment from 'moment';
+
+/**
+ * Define the anomaly action types
+ */
+export const ActionTypes = {
+  LOAD: type('[Dimensions] Load'),
+  LOADING: type('[Dimensions] Loading'),
+  LOAD_TIMESERIES: type('[Dimensions] Load TimeSeries'),
+  LOAD_HEATMAP: type('[Dimensions] Load HeatMap'),
+  SET: type('[Dimension] Set Dimension'),
+  RESET: type('[Dimensions] Reset Data'),
+  REQUEST_FAIL: type('[Dimensions] Request Fail')
+};
+
+function resetData() {
+  return {
+    type: ActionTypes.RESET
+  };
+}
+
+function loading() {
+  return {
+    type: ActionTypes.LOADING
+  };
+}
+
+function setDimension(dimension = 'All') {
+  return {
+    type: ActionTypes.SET,
+    payload: dimension
+  };
+}
+
+function loadTimeSeries(response) {
+  return {
+    type: ActionTypes.LOAD_TIMESERIES,
+    payload: response
+  };
+}
+
+function loadHeatMap(response) {
+  return {
+    type: ActionTypes.LOAD_HEATMAP,
+    payload: response
+  };
+}
+
+function load(response, metricId) {
+  return {
+    type: ActionTypes.LOAD,
+    payload: response,
+    metricId
+  };
+}
+
+function requestFail() {
+  return {
+    type: ActionTypes.REQUEST_FAIL
+  };
+}
+
+/**
+* Fetches the dimensions data for a specific metric
+* @param {Number} metricId
+*/
+function fetchDimensions(metricId) {
+  return (dispatch, getState) => {
+    const { dimensions } = getState();
+
+    if (dimensions.metricId === metricId) {
+      return {};
+    }
+
+    // TODO: save url in an API folder
+    // need to have a new endpoint with just the anomaly details
+    return fetch(`/data/autocomplete/dimensions/metric/${metricId}`)
+      .then(res => res.json())
+      .then(res => dispatch(load(res, metricId)))
+      .catch((res) => dispatch(requestFail(res)));
```

Comment:
we should also utilize your checkStatus so that i actually fails. This catch is just aesthetic for the moment.